prml/generic-asset/src/lib.rs

```diff
@@ -185,15 +185,17 @@ pub use types::*;
 use weights::WeightInfo;
 
 /// Minimal information for an account.
-#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
-pub enum AccountData {
-	None,
-	Exists,
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, Default)]
+pub struct AccountData<AssetId: Ord> {
+	/// Contains asset ids whose balances are greater than existential deposit.
+	significant_assets: BTreeSet<AssetId>,
+	/// True if the account should persist even with dust balances
+	persistent: bool,
 }
 
-impl Default for AccountData {
-	fn default() -> Self {
-		AccountData::None
+impl<AssetId: AtLeast32BitUnsigned + Copy> AccountData<AssetId> {
+	fn is_significant(&self) -> bool {
+		!self.significant_assets.is_empty() || self.persistent
```

Comment:
assuming 'persistent' is required, makes sense to short circuit
```suggestion
	fn exists(&self) -> bool {
		self.persistent || !self.significant_assets.is_empty()
```

Reply:
Will do that.